chain/rosetta-rpc/src/lib.rs

```diff
@@ -185,6 +188,80 @@ async fn network_options(
     }))
 }
 
+/// Get a block with `block_id`.
+/// Returns `Ok(Some(_))` if the block exists and is final.
+/// Returns `Ok(None)` if the block does not exist or is not final.
+async fn get_block_if_final(
+    block_id: &near_primitives::types::BlockReference,
+    view_client_addr: &Addr<ViewClientActor>,
+) -> Result<Option<near_primitives::views::BlockView>, models::Error> {
+    let final_block = get_final_block(view_client_addr).await?;
+    let is_query_by_height = match block_id {
+        near_primitives::types::BlockReference::Finality(
+            near_primitives::types::Finality::Final,
+        ) => return Ok(Some(final_block)),
+        near_primitives::types::BlockReference::BlockId(
+            near_primitives::types::BlockId::Height(height),
+        ) => {
+            if height > &final_block.header.height {
+                return Ok(None);
+            }
+            if height == &final_block.header.height {
+                return Ok(Some(final_block));
+            }
+            true
+        }
+        _ => false,
```

Comment:
Instead of a catch-all arm (`_`), prefer explicitly list all the enum variants, so in the future, when there are more variants added, Rust will force the contributor to review the unhandled conditions.